org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaProtocolExtensions.java

```diff
@@ -35,4 +38,12 @@
 	 */
 	@JsonNotification
 	void projectConfigurationUpdate(TextDocumentIdentifier documentUri);
+
+	/**
+	 * The implementors request is sent from the client to the server to request
+	 * implementors information at a given text document position.
+	 *
+	 */
+	@JsonRequest(value = "java/implementors", useSegment = false)
+	CompletableFuture<List<? extends SymbolInformation>> implementors(TextDocumentPositionParams position);
```

Comment:
We probably need something more open for the parameter, eg. what if the request only contains an FCQN? I think we should have only 1 `java/implementors` entry point, but the incoming parameter should probably wrap TextDocumentPositionParams and some other stuff. Shooting from the hip:
```
ImplementorRequestParams {
   - query {
          - fullyQualifiedName: String (optional)
   or   - position : TextDocumentPositionParams (optional) 
   }
   - options : Map<String, String> (optional)
}
```  
We need more info from @tsmaeder about potential usage of that API in Che.

Reply:
Shouldn't we call it `java/typeHierarchy` instead?